src/fs_base_setxattr.hpp

```diff
@@ -33,13 +33,16 @@ namespace fs
             const char        *name,
             const void        *value,
             const size_t       size,
-            const int          flags)
+            const int          flags,
+            const u_int32_t    position)
   {
-#ifndef WITHOUT_XATTR
-    return ::lsetxattr(path.c_str(),name,value,size,flags);
-#else
+  #if WITHOUT_XATTR
     return (errno=ENOTSUP,-1);
-#endif
+  #elif __APPLE__
+    return ::setxattr(path.c_str(),name,value,size,position,flags & XATTR_NOFOLLOW);
```

Comment:
Shouldn't `flags & XATTR_NOFOLLOW` actually be `flags | XATTR_NOFOLLOW`? I think using bitwise AND means that it will always be 0, effectively discarding the value of flags.

There's a few other similar places where bitwise AND is used where it probably should be a OR.